src/Ledger/Utxo/Properties.lagda

```diff
@@ -544,3 +544,139 @@ pov {tx}{utxo}{_}{fees}{deposits}{donations} h'
+  {- 2 -} {_∪⁺∅ᵐ∣∅ˢ≡id   : (deposits :  DepositPurpose ⇀ Coin) → getCoin deposits ≡ getCoin ((deposits ∪⁺ ∅ᵐ) ∣ ∅ˢ ᶜ)}
+  {- 3 -} {∪⁺singleton≡  : {deps : DepositPurpose ⇀ Coin} {cDep : DepositPurpose × Coin}
+                             → getCoin (deps ∪⁺ ❴ cDep ❵) ≡ getCoin deps + proj₂ cDep }
+  where
+
+  gmsc :  let open Tx tx renaming (body to txb); open TxBody txb
+              pp = UTxOEnv.pparams Γ; open PParams pp
+              open UTxOState utxoState renaming (utxo to st; fees to fs; deposits to deps; donations to dons)
+          in
+
+  {- 4 -} Γ ⊢  ⟦ st   , fs   , deps   , dons   ⟧ᵘ ⇀⦇ tx ,UTXO⦈
+               ⟦ utxo'  , fees'  , deposits'  , donations'  ⟧ᵘ
+
+  {- 5 -} →  (∀ {p} → p ∈ˡ txprop → GovProposal.deposit p ≡ govActionDeposit)
+  {- 6 -} →  noRefundCert txcerts
+             -------------------------------------------------------------------
+          →  coin (consumed pp utxoState txb) ≥ length txprop * govActionDeposit
+
+  gmsc step@(UTXO-inductive⋯ tx Γ utxoState _ _ _ _ c≡p cmint≡0 _ _ _ _ _ _ _) h nrf =
+    ≤-trans v (≤-reflexive (sym coin-hom))
+
+    where
+      pp : PParams
+      pp = UTxOEnv.pparams Γ
+      open Tx tx renaming (body to txb); open TxBody txb
+      open UTxOState utxoState renaming (utxo to st; fees to fs; deposits to deps; donations to dons)
+      open PParams pp
+
+      newDeps refunds : Coin
+      newDeps    = newDeposits pp utxoState txb
+      refunds    = depositRefunds pp utxoState txb
+
+      balIn balOut : Value
+      balIn = balance (st ∣ txins)
+      balOut = balance (outs txb)
+
+      -- The sum of all GovProposal deposits is the number of GovProposals times govActionDeposit.
+      i : sum (map GovProposal.deposit txprop) ≡ length txprop * govActionDeposit
+
+      -- The positive part of change in deposits is at least the sum of GovProposal deposits.
+      ii : newDeps ≥ sum (map GovProposal.deposit txprop)
+
+      -- Coin consumed ≡ coin produced
+      iii : coin (balIn + mint) + refunds ≡ coin balOut + txfee + newDeps + txdonation
+
+      -- Coin produced is at least the sum of GovProposal deposits.
+      iv : coin balOut + txfee + newDeps + txdonation ≥ sum (map GovProposal.deposit txprop)
+
+      -- Coin consumed is at least the number of GovProposals times govActionDeposit.
+      v : coin (balIn + mint) + refunds ≥ length txprop * govActionDeposit
+
+      -- Lemmas -------------------------------------------------------------------------
+      coin-hom : ∀ {val} {c} → coin (val + inject c) ≡ coin val + c
+      coin-hom {val} {c} = begin
+        coin (val + inject c)         ≡⟨ homo coinIsMonoidHomomorphism val (inject c) ⟩
+        coin val + (coin ∘ inject) c  ≡⟨ cong (coin val +_) (property c) ⟩
+        coin val + c                  ∎
+
+      module _ {txid : TxId} {gaDep : Coin} {deposits : DepositPurpose ⇀ Coin}
+        where
+        updatePropDeps≥ : (props : List GovProposal)
```

Comment:
Since you're already using a module to share arguments, why not `props` as well?

Reply:
Because the proof is by induction on `props`.